ui/packages/catalog-ui-search/src/main/webapp/react-component/location/gazetteer.js

```diff
@@ -86,12 +87,68 @@ class Gazetteer extends React.Component {
       }
     })
   }
+  getMaxMinOfPolygon(polygonCoordinates, isMultiPolygon) {
+    var finalMax = { x: Number.MIN_SAFE_INTEGER, y: Number.MIN_SAFE_INTEGER }
+    var finalMin = { x: Number.MAX_SAFE_INTEGER, y: Number.MAX_SAFE_INTEGER }
+    var maxArea = -1
+    var currentArea = -1
+    var currentMax
+    var currentMin
+    polygonCoordinates.map(rowCoordinates => {
+      currentMax = { x: Number.MIN_SAFE_INTEGER, y: Number.MIN_SAFE_INTEGER }
+      currentMin = { x: Number.MAX_SAFE_INTEGER, y: Number.MAX_SAFE_INTEGER }
+      if (isMultiPolygon) {
+        rowCoordinates[0].map(coordinates => {
+          currentMax.x = Math.max(coordinates[0], currentMax.x)
+          currentMax.y = Math.max(coordinates[1], currentMax.y)
+          currentMin.x = Math.min(coordinates[0], currentMin.x)
+          currentMin.y = Math.min(coordinates[1], currentMin.y)
+        })
+      } else {
+        rowCoordinates.map(coordinates => {
+          currentMax.x = Math.max(coordinates[0], currentMax.x)
+          currentMax.y = Math.max(coordinates[1], currentMax.y)
+          currentMin.x = Math.min(coordinates[0], currentMin.x)
+          currentMin.y = Math.min(coordinates[1], currentMin.y)
+        })
+      }
+      currentArea =
+        (currentMax.x - currentMin.x) * (currentMax.y - currentMin.y)
+      if (Math.max(currentArea, maxArea) > maxArea) {
+        maxArea = currentArea
+        finalMax = currentMax
+        finalMin = currentMin
+      }
+    })
+    var finalArea = {
+      maxX: finalMax.x,
+      minX: finalMin.x,
+      maxY: finalMax.y,
+      minY: finalMin.y,
+    }
+    return finalArea
+  }
   async geofeature(suggestion) {
     const [type, id] = suggestion.id.split(':')
     const res = await window.fetch(
-      `https://nominatim.openstreetmap.org/reverse?format=json&osm_type=${type}&osm_id=${id}`
+      `https://nominatim.openstreetmap.org/reverse?format=json&osm_type=${type}&osm_id=${id}&polygon_geojson=1`
     )
     const data = await res.json()
+    if (Object.keys(data.address).length === 2) {
+      var finalArea
+      if (
+        data.geojson.coordinates[0][0][0] &&
+        !data.geojson.coordinates[0][0][0][0]
```

Comment:
If you extract this logic into its own `isMultiPolygon` function, you can remove this if/else branch and do `const finalArea = this.getMaxMinOfPolygon(data.geojson.coordinates, this.isMultiPolygon(data.geojson.coordinates))`